src/main/java/io/stargate/sgv2/jsonapi/service/projection/ProjectionPath.java

```diff
@@ -14,9 +14,10 @@
  * ampersands within segments.
  *
  * <p>Use {@link #from(String)} to create an instance, {@link #getSegment(int)} to access the
- * segments, and {@link #encodeNoEscaping()} to get the dot-separated path string.
+ * segments, and {@link #encodeSegment(String)} to get the dot-separated path string.
  */
 public final class ProjectionPath implements Comparable<ProjectionPath> {
+  private final String path;
```

Comment:
Let's rename as `encodedPath` -- or add a comment mentioning it is full encoded path.

Would actually make sense to also add comment next to `segment` to mention that segment Strings are not escaped.